File: scripts/extract_tardis_html.py
import json
import sys
from bs4 import BeautifulSoup

html = open(sys.argv[1], encoding='utf-8').read()
soup = BeautifulSoup(html, 'html.parser')
links = []
for el in soup.find_all('a'):
    links.append({'text': ' '.join(el.get_text(' ', strip=True).split()), 'href': el.get('href', '')})
images = [{'alt': img.get('alt', ''), 'src': img.get('src', '')} for img in soup.find_all('img')]
cards = []
for el in soup.find_all(['article', 'div']):
    classes = ' '.join(el.get('class', []))
    text = ' '.join(el.get_text(' ', strip=True).split())
    if text and ('card' in classes.lower() or el.name == 'article'):
        cards.append({'text': text, 'html': str(el)[:2000]})
out = {'links': links, 'images': images, 'cards': cards}
open('/home/ubuntu/novelas-tv/tardis-home-links.json', 'w', encoding='utf-8').write(json.dumps(out, ensure_ascii=False, indent=2))
print(json.dumps({'linkCount': len(links), 'imageCount': len(images), 'cardCount': len(cards)}, indent=2))

File: scripts/extract_tardis_html.mjs
import fs from 'node:fs';
import { load } from 'cheerio';

const input = process.argv[2];
const html = fs.readFileSync(input, 'utf8');
const $ = load(html);
const links = [];
$('a').each((_, el) => {
  const text = $(el).text().replace(/\s+/g, ' ').trim();
  const href = $(el).attr('href') || '';
  if (text || href) links.push({ text, href });
});
const images = [];
$('img').each((_, el) => {
  images.push({ alt: $(el).attr('alt') || '', src: $(el).attr('src') || '' });
});
const cards = [];
$('[class*="card"], article').each((_, el) => {
  const text = $(el).text().replace(/\s+/g, ' ').trim();
  if (text) cards.push({ text, html: $(el).html()?.slice(0, 2000) || '' });
});
fs.writeFileSync('/home/ubuntu/novelas-tv/tardis-home-links.json', JSON.stringify({ links, images, cards }, null, 2));
console.log(JSON.stringify({ linkCount: links.length, imageCount: images.length, cardCount: cards.length }, null, 2));
